activerecord/lib/active_record/explain.rb

```diff
@@ -50,5 +50,21 @@ def render_bind(attr)
 
         [attr&.name, value]
       end
+
+      def build_explain_clause(options = [])
+        if connection.respond_to?(:build_explain_clause)
+          connection.build_explain_clause(options)
+        else
+          "EXPLAIN for:"
```

Comment:
We should show a deprecation here so the adapter implement this new method. I don't want to support this branch forever.